File: class0403/Project1/Project1/Item.cpp
#include "Item.h"
#include <iostream>
#include <fstream>
#include <string>
#include <iomanip>

using namespace std;

int Item::serialNo = 1;

void Item::read(ifstream& fin) {
	id = serialNo++;
	fin >> name >> price >> type;
}
void Item::print() {
	string t = "Man";
	if (type == "f") t = "Woman";
	else if (type == "c") type = "Common";
	cout << id << ' ' << name << '\t' << setw(6)
		<< price << "원 " << type<<endl;
}
void Item::printResult() {
	string t = "Man";
	if (type == "f") t = "Woman";
	else if (type == "c") type = "Common";
	cout << id << ' ' << name << '\t' << setw(6)
		<< price << "원 " << type << "\t총 합계:" << sales * price << endl;
}




//생성자
Item::Item()
{
	sales = 0;
	cout << "생성!" << endl;
	
}

//소멸자
Item::~Item()
{
}

File: class0403/0403JAVA/src/main/Item.java
package main;

import java.util.Scanner;

public class Item {
    static int serialNo = 1;
    int id;
    String name;
    String type;
    int price;
    int order_num;

    void read(Scanner scan) {
        id = serialNo++;
        name = scan.next();
        price = scan.nextInt();
        type = scan.next();
    }

    void print() {
        String type = "Man";
        if (type.equals("f")) type = "Woman";
        else if (type.equals("c")) type = "Common";
        System.out.printf("[%d] %s %s ", id, name, type);
        System.out.print(price + "원\n");
    }
}
